scripts/manageacl.py

```diff
@@ -0,0 +1,158 @@
+import argparse
+import asyncio
+from enum import Enum
+
+from azure.core.credentials import AzureKeyCredential
+from azure.core.credentials_async import AsyncTokenCredential
+from azure.identity import AzureDeveloperCliCredential
+from azure.search.documents.aio import SearchClient
+from azure.search.documents.indexes.aio import SearchIndexClient
+from azure.search.documents.indexes.models import SearchFieldDataType, SimpleField
+
+class ManageAcl:
+    async def __aenter__(self):
+        endpoint = f"https://{self.service_name}.search.windows.net"
+        self.search_client = SearchClient(endpoint=endpoint, credential=self.credentials, index_name=self.index_name)
+        await self.search_client.__aenter__()
```

Comment:
it might look nicer to use "async with search_client" here? We usually avoid direct calling dunders when possible, prefer syntactic sugar :)